benchmark/cublas/utils.hpp

```diff
@@ -274,6 +276,23 @@ static inline std::tuple<double, double> timef_cuda(function_t func,
   return std::make_tuple(overall_time, static_cast<double>(elapsed_time) * 1E6);
 }
 
+/**
+ * Reference type of the underlying benchmark data aimed to match the
+ * cuda/cuBLAS scalar types.
+ */
+template <typename T, typename Enable = void>
+struct CudaType {
+  using type = T;
+};
+
+#ifdef BLAS_ENABLE_HALF
+// When T is sycl::half, use cuda's __cuda as type.
+template <typename T>
+struct CudaType<T, std::enable_if_t<std::is_same_v<T, cl::sycl::half>>> {
+  using type = __half;
+};
+#endif
```

Comment:
```suggestion
// When T is sycl::half, use cuda's __cuda as type.
template <typename T>
struct CudaType<T, std::enable_if_t<std::is_same_v<T, cl::sycl::half>>> {
  using type = __half;
};
```
I think it is possible to remove also this guard.